xrpackage-engine.js

```diff
@@ -0,0 +1,1058 @@
+  result.leftOffset = _makeTypedArray(Float32Array, 3);
+  result.leftOffset.set(Float32Array.from([-0.625/2, 0, 0]));
+  result.rightOffset = _makeTypedArray(Float32Array, 3);
+  result.leftOffset.set(Float32Array.from([0.625/2, 0, 0]));
+  result.leftFov = _makeTypedArray(Float32Array, 4);
+  result.leftFov.set(Float32Array.from([45, 45, 45, 45]));
+  result.rightFov = _makeTypedArray(Float32Array, 4);
+  result.rightFov.set(result.leftFov);
+  const _makeGamepad = () => ({
+    connected: _makeTypedArray(Uint32Array, 1),
+    position: _makeTypedArray(Float32Array, 3),
+    orientation: (() => {
+      const result = _makeTypedArray(Float32Array, 4);
+      result[3] = 1;
+      return result;
+    })(),
+    direction: (() => { // derived
+      const result = _makeTypedArray(Float32Array, 4);
+      result[2] = -1;
+      return result;
+    })(),
+    transformMatrix: (() => { // derived
+      const result = _makeTypedArray(Float32Array, 16);
+      result.set(Float32Array.from([1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1]));
+      return result;
+    })(),
+    buttons: (() => {
+      const result = Array(10);
+      for (let i = 0; i < result.length; i++) {
+        result[i] = {
+          pressed: _makeTypedArray(Uint32Array, 1),
+          touched: _makeTypedArray(Uint32Array, 1),
+          value: _makeTypedArray(Float32Array, 1),
+        };
+      }
+      return result;
+    })(),
+    axes: _makeTypedArray(Float32Array, 10),
+  });
+  result.gamepads = (() => {
+    const result = Array(2);
+    for (let i = 0; i < result.length; i++) {
+      result[i] = _makeGamepad();
+    }
+    return result;
+  })();
+
+  return result;
+};
+
+const _setFramebufferMsRenderbuffer = (gl, xrfb, width, height, devicePixelRatio) => {
+  const oldDrawFbo = gl.getParameter(gl.DRAW_FRAMEBUFFER_BINDING);
+  const oldRbo = gl.getParameter(gl.RENDERBUFFER_BINDING);
+
+  const colorRenderbuffer = gl.createRenderbuffer();
+  const depthRenderbuffer = gl.createRenderbuffer();
+  const colorTex = gl.createTexture();
+  const depthTex = gl.createTexture();
+
+  gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, xrfb);
+
+  gl.bindRenderbuffer(gl.RENDERBUFFER, colorRenderbuffer);
+  gl.renderbufferStorageMultisample(gl.RENDERBUFFER, 4, gl.RGBA8, width * devicePixelRatio, height * devicePixelRatio);
+  gl.framebufferRenderbuffer(gl.DRAW_FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.RENDERBUFFER, colorRenderbuffer);
+
+  gl.bindRenderbuffer(gl.RENDERBUFFER, depthRenderbuffer);
+  gl.renderbufferStorageMultisample(gl.RENDERBUFFER, 4, gl.DEPTH32F_STENCIL8, width * devicePixelRatio, height * devicePixelRatio);
+  gl.framebufferRenderbuffer(gl.DRAW_FRAMEBUFFER, gl.DEPTH_STENCIL_ATTACHMENT, gl.RENDERBUFFER, depthRenderbuffer);
+
+  gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, oldDrawFbo);
+  gl.bindRenderbuffer(gl.RENDERBUFFER, oldRbo);
+};
+
+export class XRPackageEngine extends EventTarget {
+  constructor (options) {
```

Comment:
Are these extra spaces following linting rules?

Reply:
I believe so!

Reply:
Can we change that? It seems another of those cases where every file in our repos will conflict because of it.